docs/faq.rst

```diff
@@ -493,6 +493,37 @@ maybe open an issue in their issue tracker. Do not file an issue in the
 If you can reproduce the issue with the proven filesystem, please file an
 issue in the |project_name| issue tracker about that.
 
+Why does running 'borgbackup check --repair' warn about data loss?
+------------------------------------------------------------------
+
+Due to the nature of backup, and especially deduplicated backups as used
+by |project_name|, some repair operations may require a roll-back to an
+earlier version of the repository. Some errors, especially where data
+corruption is involved, may require purging corrupted data to bring the
+repository back to a known good state.
+
+Also, we can't predict every possible corruption vector for a repository.
+Some forms of corruption may be undetected, and some attempts to repair
+corruption could break other parts of the repository.
+
+Repair is considered a destructive operation in that it will modify the
+repository in order to automatically repair some known methods of corruption.
+
+In other words, the warning is there to emphasize that |project_name| will:
+  - Perform automated routines that will modify your backup repository
+  - May not actually fix the problem you are experiencing
+  - May further corrupt your repository
+  - May cause other problems which may not be noticable
```

Comment:
e.g. a corrupted or modified document segment stored in the repo. Assuming --repair will modify segment files if it detects they may be corrupted.

None of this is likely ... but the basis of this entire FAQ entry is to try to explain why the warning is displayed, so I'm trying to expand on that and give all the real world problems that could possibly be caused by using an automated repair routine...

All of this, I understand, is unlikely. I debated even taking on this task because I hoped the warning would eventually go away once you had more confidence that it had been used and tested more thoroughly. Once I saw that repair was implemented 3 years ago, it changed my perspective of why the warning existed.